inst/shiny/modules/tab_visuals.R

```diff
@@ -0,0 +1,560 @@
+        choices = intersect(names(boxplotdata()), names(data())),
+        selected = "DOSEA"
+      )
+    })
+
+    output$select_colorvars_boxplot <- renderUI({
+      pickerInput(
+        inputId = ns("selected_colorvars_boxplot"),
+        label = "Select coloring variables to differentiate boxplots",
+        multiple = TRUE,
+        choices = intersect(names(boxplotdata()), names(data())),
+        selected = "DOSNO"
+      )
+    })
+
+    observeEvent(list(input$selected_xvars_boxplot,
+                      input$selected_colorvars_boxplot), {
+                   xvar_options_list <- lapply(
+                     c(input$selected_xvars_boxplot, input$selected_colorvars_boxplot),
+                     \(id_var) paste(id_var, unique(boxplotdata()[[id_var]]), sep = ": ")
+                   )
+
+                   names(xvar_options_list) <- c(
+                     input$selected_xvars_boxplot,
+                     input$selected_colorvars_boxplot
+                   )
+
+                   updatePickerInput(
+                     session = session,
+                     inputId = "selected_varvalstofilter_boxplot",
+                     label = "Select values to display for grouping",
+                     choices = xvar_options_list,
+                     selected = unlist(xvar_options_list)
+                   )
+                 })
+
+    # toggle between boxplot and violinplot
+    output$violin_toggle <- renderUI({
+      switchInput(
+        inputId = ns("violinplot_toggle_switch"),
+        label = "",
+        value = TRUE,
+        onLabel = "Boxplot",
+        offLabel = "Violinplot"
+      )
+    })
+
+    # compute the box plot
+    output$boxplot <- renderPlotly({
+      req(boxplotdata())
+      req(input$selected_param_boxplot)
+      req(input$selected_xvars_boxplot)
+      req(input$selected_colorvars_boxplot)
+      req(input$selected_varvalstofilter_boxplot)
+
+      flexible_violinboxplot(
+        boxplotdata = boxplotdata(),
+        parameter = input$selected_param_boxplot,
+        xvars = input$selected_xvars_boxplot,
+        colorvars = input$selected_colorvars_boxplot,
+        varvalstofilter = input$selected_varvalstofilter_boxplot,
+        columns_to_hover = unname(unlist(res_nca()$data$conc$columns$groups)),
+        box = input$violinplot_toggle_switch
+      )
+    })
+
+    # EXPORT Report ------------------------------------------------------------
+
+    # ATTENTION: most of the blocks in the RMD are still eval = FALSE, as
+    # feature is under development and only implemented as preview for idea of
+    # exporting reports
+
+    # templates from: https://opensource.nibr.com/xgx/
+
+    # concat all dataframes into one object
+    concat_report_data <- reactive({
+      data <- res_nca()
+      data$formatted_res <- boxplotdata()
+      data
+    })
+
+    # render the markdown upon clicking generate
+    rendered_rmd <- reactiveVal(NULL)
+
+    observeEvent(input$generate_report, {
```

Comment:
issue: Seems to be related to the commented out code. As of now it is dead code, should be removed / moved alongside the commented out parts.